src/test/php/org/bovigo/vfs/vfsStreamWrapperFileTestCase.php

```diff
@@ -454,4 +454,22 @@ public function cannotReadFileFromNonReadableDir()
         $this->bar->chmod(0);
         $this->assertFalse(@file_get_contents($this->baz1URL));
     }
+
+    /**
+     * @test
+     * @see   https://github.com/mikey179/vfsStream/issues/155
+     * @since 1.6.5
+     */
+    public function errorOnOpen()
+    {
+        $file = $this->baz1;
+        $path = $file->url();
+        $this->assertNotFalse(@fopen($path, 'r'), 'precondition');
+
+        $file->setOpenError("fopen() gives error");
+        $this->assertFalse(@fopen($path, 'r'), 'fopen fails now');
+
+        $file->setOpenError(null);
+        $this->assertNotFalse(@fopen($path, 'r'), 'fopen works again');
+    }
 }
```

Comment:
Well, I should have talked about behaviour instead, I guess... I think there are three different behaviours covered in this test, and each behaviour should have its own test:
1. What happens when no open error is set
2. What happens when an open error is set
3. What happens when an open error is reset

When you look at it like this you automatically end with one assertion per test again. Having more than one assertion per test can be fine, but generally I think it is a potential code smell that the tests could be structured in a better way. These three different behaviours also give a good indication how the according test methods should be named.

On the other hand you are right, this discussion might become moot depending on the other part of the discussion. :)